### ISSUE TITLE
Removing a deferred filter via the filter indicators bar has unexpected results

### ISSUE THREAD

**Issue by austincarpenter** [laravel-filament/filament#10831]
### Package

filament/filament

### Package Version

v3.2

### Laravel Version

v10.40

### Livewire Version

v3.3.5

### PHP Version

8.3

### Problem description

If filters are deferred, clicking on their respective 'x' buttons in the filter indicators bar does not immediately remove them. I understand this is what the deferring of the filters does, however with no apply filters button visible in the bar, this behaviour is confusing.

### Expected behavior

A few options:
1. Interacting with filters (and the remove all button) in the filter indicators bar should have immediate effect, regardless of whether the main filters panel defers applying them or not.
2. The filter should remain applied upon clicking the 'x' and the badge should actually disappear (it doesn't currently).
3. Hiding the filter indicators altogether when deferring

I would expect it to function as per the first option – this way we get the best of both worlds. The second I would find confusing and the third means trading one feature for another. Just my opinion though!

### Steps to reproduce

1. Apply `->deferFilters()`
2. Apply a filter
3. Attempt to remove it in the active filters bar

### Reproduction repository

https://github.com/filamentphp/demo with standard filter layout in ProductResource

**Comment by MhdTarhini** [comment]
Hey, I'm facing similar issue,  when I click on remove all filters an error message "foreach() argument must be of type array|object, null given" appear,  I things this error is referring to $filterFields in removeTableFilter() function.
how I can manage this issues ?
Thank you

**Comment by V13Axel** [comment]
@MhdTarhini Dunno if you ever solved that, but I found this thread when I encountered a similar issue. Thought I'd leave an answer here for anyone else that hits it:

My problem turned out to be that I was creating my filters via `->filters()` as usual, then calling `->filtersFormSchema()` to arrange them into a desired schema... but not using them all. Any unused filters caused `removeTableFilter()` to choke on a null instead of an array.

Using all of them in the schema fixed it.
